contracts/UniswapUSDCAirdrop.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.8.15;
+
+import "./MerkleDistributor.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract UniswapUSDCAirdrop is MerkleDistributor, Ownable {
+    uint256 public immutable endTime;
+
+    constructor (address token_, bytes32 merkleRoot_, uint256 endTime_) MerkleDistributor (token_, merkleRoot_) {
+        endTime = endTime_;
+    }
+
+    function claim(uint256 index, address account, uint256 amount, bytes32[] calldata merkleProof) public override {
+        require(block.timestamp < endTime, "Claim window is finished");
+        super.claim(index, account, amount, merkleProof);
+    }
+
+    function withdraw() external onlyOwner {
+        require(block.timestamp >= endTime, "Cannot withdraw during claim window");
+        IERC20(token).transfer(msg.sender, IERC20(token).balanceOf(address(this)));
```

Comment:
s/safeTransfer? And in MerkleDistributor -- doesn't matter for USDC but unsafe for USDT for example

Reply:
OZ SafeTransfer takes care of USDT approval issue right?

Reply:
To confirm you mean using `safeTransfer()` from https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#SafeERC20 ? For our purposes its unnecessary as we use USDC but I could just use SafeERC20 anyways.

Reply:
Yeah, I think it would be worth it esp if other people are going to fork and use for general merkle tree airdrops

Reply:
Yeah exactly, its a hidden gotcha if this is used for other tokens. We also have https://github.com/Uniswap/v3-periphery/blob/main/contracts/libraries/TransferHelper.sol fwiw